### DISCUSSION TITLE
Linking to a temporary codespace for manual validation

### DISCUSSION THREAD

**Discussion by anthony-c-martin** [community/community#17289] [Codespaces]
I work on a CLI tool and VSCode extension hosted on GitHub. Our CI process will build both when a user submits a PR. Often a reviewer will want to play around interactively to validate the new feature behaves as expected.

It would be **very** cool if we could automatically post a link in each PR to spin up a Codespaces instance with the CI-built CLI + extension installed, or provide a bash script to do something similar. This would make manually validating changes super easy. Has anyone tried anything like this?

**Comment by legomushroom** [comment]
@anthony-c-martin we have a similar setup for the https://github/github/gh-net CLI extension. 

1. We maintain a makefile that is able to install a CLI extension on the current machine (build it and move to the extensions folder of the GitHub CLI). Hence anyone can open PR in a Codespace and run the script to install/test it there (or do the same on the local machine).

2. When a certain tag is pushed (PR can be another trigger for this) - CI builds the CLI extension and publishes it to a public "beta" repo for tests, in my case https://github.com/legomushroom/gh-net that can also be used for tests.

I assume the same could be done for the vscode extension too 🤔

**Reply by anthony-c-martin** [reply]
Thanks for sharing! Do you have a pointer to the makefile/script by any chance? Would love to see how this is implemented.
